op-service/client/http.go

```diff
@@ -23,16 +23,32 @@ type HTTP interface {
 
 type BasicHTTPClient struct {
 	endpoint string
-	log      log.Logger
-	client   *http.Client
+	header   http.Header // optional header to use in every request
+
+	log    log.Logger
+	client *http.Client
 }
 
-func NewBasicHTTPClient(endpoint string, log log.Logger) *BasicHTTPClient {
-	return &BasicHTTPClient{
+func NewBasicHTTPClient(endpoint string, log log.Logger, opts ...BasicHTTPClientOption) *BasicHTTPClient {
+	c := &BasicHTTPClient{
 		endpoint: endpoint,
 		log:      log,
 		client:   &http.Client{Timeout: DefaultTimeoutSeconds * time.Second},
 	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+type BasicHTTPClientOption = func(*BasicHTTPClient)
+
+func WithHeader(h http.Header) BasicHTTPClientOption {
+	return func(c *BasicHTTPClient) {
+		c.header = h
+	}
 }
```

Comment:
I'd prefer if we define `Option` as interface with an `Apply` method, and then define a `HttpOptionFn` type with a method that implements the `Apply` by calling the function. This is what the Go standard lib does for http function handlers, and what we generally try to do for option patterns in the code-base.
It also helps to type the option explicitly that way, rather than just using a function `=` alias that leaves it untyped.